lib/host.js

```diff
@@ -127,6 +131,11 @@ Host.prototype.canBeConsideredAsUp = function () {
 Host.prototype.setDistance = function (distance) {
   distance = distance || types.distance.local;
   this.pool.coreConnectionsLength = this.options.pooling.coreConnectionsPerHost[distance];
+  //if (distance === types.distance.ignored && this.pool.connections.length > 0) {
```

Comment:
I don't know why I commented it out, it makes sense: a custom load balancing policy may determine that a host that was being used now should be ignored.